src/proc.py

```diff
@@ -29,16 +29,27 @@
 from subprocess import * # flake8: noqa
 
 
-def FixPython(cmd):
+def Which(filename, cwd):
+  to_search = ([cwd] + os.environ['PATH'].split(os.pathsep))
+  for path in to_search:
+    abs_path = os.path.join(path, filename)
+    if os.path.exists(abs_path):
```

Comment:
[Huh, TIL.](http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html#tag_08_03)

Also TIL, [bash returns the first file it sees in your PATH if none are executable](http://git.savannah.gnu.org/cgit/bash.git/tree/findcmd.c#n553), presumably to give you a better error message? `which` is more directly standards-compliant in that it doesn't return anything when you ask `which non-executable-file-in-path`.